hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/AbstractContractCopyTest.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.contract;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.junit.Test;
+
+import static org.apache.hadoop.fs.CommonPathCapabilities.FS_NATIVE_COPY;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.createFile;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.dataset;
+import static org.apache.hadoop.fs.contract.ContractTestUtils.touch;
+import static org.apache.hadoop.test.LambdaTestUtils.intercept;
+
+/**
+ * Test native copy - if supported.
+ */
+public abstract class AbstractContractCopyTest extends AbstractFSContractTestBase {
+
+  protected Path srcDir;
+  protected Path srcFile;
+  protected Path zeroByteFile;
+  protected Path dstDir;
+  protected Path dstFile;
+  protected FileSystem fs;
+
+  @Override
+  public void setup() throws Exception {
+    super.setup();
+    srcDir = absolutepath("source");
+    dstDir = absolutepath("target");
+
+    srcFile = new Path(srcDir, "source.txt");
+    dstFile = new Path(dstDir, "dst.txt");
+    zeroByteFile = new Path(srcDir, "zero.txt");
+
+    byte[] block = dataset(TEST_FILE_LEN, 0, 255);
+    createFile(getFileSystem(), srcFile, true, block);
+    touch(getFileSystem(), zeroByteFile);
+
+    fs = getFileSystem();
+
+    skipIfUnsupported(SUPPORTS_NATIVE_COPY);
+  }
+
+  @Override
+  public void teardown() throws Exception {
+    describe("\nTeardown\n");
+    super.teardown();
+  }
+
+
+  @Test
+  public void testBasicCopy() throws Throwable {
+    describe("Checks whether S3A FS supports native copy"
+        + " and copies a file from one location to another."
+        + " File deletion taken care in teardown.");
+
+    assertPathExists(srcFile + " does not exist", srcFile);
+
+    fs.delete(dstFile, false);
+    // initiate copy
+    fs.copyFile(srcFile.toUri(), dstFile.toUri());
+
+    // Check if file is available
+    assertIsFile(dstFile);
```

Comment:
you need to 
* actually verify the data
* assert that the source file is stil there